src/observers/form_link_click_observer.ts

```diff
@@ -53,7 +53,10 @@ export class FormLinkClickObserver implements LinkClickObserverDelegate {
     if (turboAction) form.setAttribute("data-turbo-action", turboAction)
 
     const turboConfirm = link.getAttribute("data-turbo-confirm")
-    if (turboConfirm) form.setAttribute("data-turbo-confirm", turboConfirm)
+    if (turboConfirm) {
+      form.setAttribute("data-turbo-confirm", turboConfirm)
+      form.originalElement = link
```

Comment:
I'm unsure about unintended side effects of introducing a property on `Element`. Is there another place we could stash this reference?

Reply:
I couldn't think of any other good place to store the reference. Open to suggestions.

Might be better to have a turbo specific name here like `turboConfirmElement` so that it wouldn't clash with any other properties?

Reply:
Here's an idea to resolve this: https://github.com/excid3/turbo/pull/2

Reply:
What do you think about actually writing to the `SubmitEvent.submitter` property?

```ts
form.addEventListener("submit", (event: SubmitEvent) => event.submitter = link, { once: true }
```

Reply:
That might even be cleaner, given you can override `submitter` and it's not being lost in-between.

Reply:
If that doesn't work, you could try passing it to [HTMLFormElement.requestSubmit](https://developer.mozilla.org/en-US/docs/Web/API/HTMLFormElement/requestSubmit#parameters) as an argument:

```ts
requestAnimationFrame(() => form.requestSubmit(link))
```

If `requestSubmit` rejects the `link` argument because it's an `HTMLAnchorElement`, and the only reason we're passing it along is to make its content and attributes available to the confirmation callback, what do you think about creating a `<button>` element that mimics the `<a>`?


```ts
const submitter = document.createElement("button")
submitter.textContent = link.textContent

for (const {name, value} of link.attributes) {
  button.setAttribute(name, value)
}

requestAnimationFrame(() => form.requestSubmit(button))
```

Reply:
If browser currently allow this, but start enforcing the types in the future, this could break (and possibly only in some browsers).

@marcoroth's idea of adding attributes and a querySelector seems like a safer solution?

Reply:
@excid3 is https://github.com/hotwired/turbo/pull/856#discussion_r1091943137 a future-compatible technique?

Its approach matches the style from the rest of the transformation logic, and it circumvents the need to stash-then-read.

Reply:
I think so, but it still lose access to the original element that way. 

It does solve the problem of passing the attributes along, so it's probably good enough™. I can't foresee a reason to need the original element personally.

I'll refactor to use that.